### PR TITLE
Parse version from GitHub tag name instead of release name 

### PR DESCRIPTION
**Pull request by huacnlee**
If we not change this, some release will parse error.

https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28

```json
{
  "name": {
    "type": [
      "string",
      "null"
    ]
  }
}
```

image

### CODE REVIEW

**Comment by huacnlee** [comment]
@cla-bot check

File: crates/util/src/github.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@ pub struct GitHubLspBinaryVersion {
 
 #[derive(Deserialize, Debug)]
 pub struct GithubRelease {
+    #[serde(rename = "tag_name")]
```
[the remark this answers is not in the document]
Comment (huacnlee) [reply crates/util/src/github.rs]: I have updated

File: crates/zed/src/languages/go.rs [hunk 2]
```diff
@@ -46,11 +46,11 @@ impl super::LspAdapter for GoLspAdapter {
         delegate: &dyn LspAdapterDelegate,
     ) -> Result<Box<dyn 'static + Send + Any>> {
         let release = latest_github_release("golang/tools", false, delegate.http_client()).await?;
-        let version: Option<String> = release.name.strip_prefix("gopls/v").map(str::to_string);
+        let version: Option<String> = release.tag_name.strip_prefix("gopls/v").map(str::to_string);
```
Comment (huacnlee) [inline crates/zed/src/languages/go.rs:49]: https://github.com/golang/tools/releases/tag/gopls%2Fv0.14.2

Here is ok, `tag_name` in `golang/tools` is `gopls/v0.14.2`

File: crates/zed/src/languages/elixir.rs [hunk 3]
```diff
@@ -112,21 +112,16 @@ impl LspAdapter for ElixirLspAdapter {
...
         let http = delegate.http_client();
         let release = latest_github_release("elixir-lsp/elixir-ls", false, http).await?;
-        let version_name = release
-            .name
-            .strip_prefix("Release ")
-            .context("Elixir-ls release name does not start with prefix")?
-            .to_owned();
 
-        let asset_name = format!("elixir-ls-{}.zip", &version_name);
+        let asset_name = format!("elixir-ls-{}.zip", &release.tag_name);
```
Comment (huacnlee) [inline crates/zed/src/languages/elixir.rs:116]: https://github.com/elixir-lsp/elixir-ls/releases/tag/v0.19.0

File: crates/zed/src/languages/go.rs [hunk 4]
```diff
@@ -46,11 +46,11 @@ impl super::LspAdapter for GoLspAdapter {
...
         if version.is_none() {
             log::warn!(
                 "couldn't infer gopls version from github release name '{}'",
```
Comment (maxdeviant) [inline crates/zed/src/languages/go.rs:52]: Could we update this log message as well to reflect the changes?

```suggestion
                "couldn't infer gopls version from github release tag name '{}'",
```

**Review by SomeoneToIgnore — APPROVED** [review]

**Pull request merged**